charmtools/charms.py

```diff
@@ -696,6 +712,33 @@ def validate_storage(charm, linter):
             linter.err('storage.{}: {}'.format(k, v))
 
 
+def validate_devices(charm, linter):
+    """Validate devices configuration in charm metadata.
+
+    :param charm: dict of charm metadata parsed from metadata.yaml
+    :param linter: :class:`CharmLinter` object to which info/warning/error
+        messages will be written
+
+    """
+    devices = charm.get('devices', None)
+    if devices is None:
+        return
+
+    if not isinstance(devices, dict) or not devices:
```

Comment:
I don't think an empty set of devices should be an error. Typically, the section would be omitted entirely, but there could be cases where it gets included as an empty dictionary due to a build processes.  An empty dictionary should be treated the same as an omitted section.

Reply:
yes, you are right.